curated_transformers/models/attention.py

```diff
@@ -73,6 +73,16 @@ def apply_causal_mask(input: Tensor) -> Tensor:
     return torch.where(causal_mask, input, blocked_value)
 
 
+class QkvHeadSharing(IntEnum):
+    """Sharing of head parameters."""
+
+    #: No parameters are shared between heads.
+    NONE = 0
+
+    #: Key shares heads, value shares heads, query has separate heads.
+    KEY_VALUE = 1
```

Comment:
We could also mention [the paper](https://arxiv.org/abs/1911.02150) here.

Reply:
Added.